Explorer/Assets/DCL/LOD/Pools/LODAsset.cs

```diff
@@ -1,100 +1,145 @@
             ProfilingCounters.LOD_Per_Level_Values[LodKey.Level].Value++;
         }
 
-        public LODAsset(LODKey lodKey)
+        public LODAsset(LODKey lodKey, ILODAssetsPool pool)
         {
-            ProfilingCounters.LODAssetAmount.Value++;
-            ProfilingCounters.Failling_LOD_Amount.Value++;
+            State = LOD_STATE.FAILED;
             LodKey = lodKey;
+            Pool = pool;
+
             Slots = Array.Empty<TextureArraySlot?>();
-            LoadingFailed = true;
             Root = null;
+            AsyncInstantiation = null;
             AssetBundleReference = null;
-            Pool = null;
+
+            ProfilingCounters.LODAssetAmount.Value++;
+            ProfilingCounters.Failling_LOD_Amount.Value++;
         }
 
         public void Dispose()
         {
-            if (LoadingFailed) return;
+            if (State == LOD_STATE.FAILED) return;
 
             AssetBundleReference.Dereference();
             AssetBundleReference = null;
 
-            if (!LodKey.Level.Equals(0))
-            {
-                for (int i = 0; i < Slots.Length; i++)
-                    Slots[i]?.FreeSlot();
-            }
-
+            for (int i = 0; i < Slots.Length; i++)
+                Slots[i]?.FreeSlot();
 
-            UnityObjectUtils.SafeDestroy(Root);
+            AsyncInstantiation.Cancel();
+            if (State == LOD_STATE.SUCCESS)
+                UnityObjectUtils.SafeDestroy(Root);
 
             ProfilingCounters.LODAssetAmount.Value--;
             ProfilingCounters.LODInstantiatedInCache.Value--;
         }
 
         public void EnableAsset()
         {
-            ProfilingCounters.LODInstantiatedInCache.Value--;
-            ProfilingCounters.LOD_Per_Level_Values[LodKey.Level].Value++;
-            Root.SetActive(true);
+            if (State == LOD_STATE.FAILED)
+                return;
+
+            if (State == LOD_STATE.SUCCESS)
+            {
+                Root.SetActive(true);
+                ProfilingCounters.LODInstantiatedInCache.Value--;
+                ProfilingCounters.LOD_Per_Level_Values[LodKey.Level].Value++;
+            }
         }
 
         public void DisableAsset()
         {
-            if (LoadingFailed) return;
-
-            ProfilingCounters.LODInstantiatedInCache.Value++;
-            ProfilingCounters.LOD_Per_Level_Values[LodKey.Level].Value--;
-
+            if (State == LOD_STATE.FAILED)
+                return;
+            
             // This logic should not be executed if the application is quitting
             if (UnityObjectUtils.IsQuitting) return;
 
-            Root.SetActive(false);
+            if (State == LOD_STATE.SUCCESS)
+            {
+                Root.SetActive(false);
+                ProfilingCounters.LODInstantiatedInCache.Value++;
+                ProfilingCounters.LOD_Per_Level_Values[LodKey.Level].Value--;
+            }
+        }
+
+        public void FinalizeInstantiation()
```

Comment:
Let's move this method to the system or extensions as well

Reply:
Suggestions applied. Moved logic to System